### ISSUE TITLE
react-collapse incompatible with preact-compat

### ISSUE THREAD

**Issue by zenfe** [developit/preact#154]
I tried to use react-collapse which uses react-motion for the transition. I get the following error:
```
Uncaught TypeError: this.props.children is not a function
```

I traced the issue down to the following piece of code:
h.js line 43 - 54:
```
		for (let j=0; j<arr.length; j++) {
			let child = arr[j],
				simple = !falsey(child) && !(child instanceof VNode);
			if (simple) child = String(child);
			if (simple && lastSimple) {
				children[children.length-1] += child;
			}
			else if (!falsey(child)) {
				children.push(child);
			}
			lastSimple = simple;
		}
```

To use react-motion the child of the Motion component is a function and preact expects a VNode or string. Otherwise it will cast the child to a String. This causes the TypeError above as the function is cast to a String.

I fixed the error by changing the above code in h.js to ensure that simple is false if child is a function.

**Comment by developit** [comment]
Nice catch, and that seems like a good fix. Did allowing the function as a child mess anything else up internally, or is this now working after that change?

**Comment by zenfe** [comment]
I thought this solved the issue but the collapse component has other issues running with preact.

Can you try to get this codepen working with preact:
https://codepen.io/nkbt/pen/MarzEg

**Comment by developit** [comment]
Will do, back from vacation this evening.

**Comment by developit** [comment]
I've released the function child support in `5.0.0-beta9`.  I'm still hoping to look into the remainder of the react-collapse issue!
